src/main/java/org/opentripplanner/common/model/P3.java

```diff
@@ -0,0 +1,30 @@
+package org.opentripplanner.common.model;
+
+/**
+ * An ordered objects of three (the same type)
+ *
+ * @param <E>
+ */
+public class P3<E> extends T3<E, E, E> {
```

Comment:
Again, this would a great candidate for a `record`.

The custom constructors can be static methods called `of`.